Description: Add EOF test to test_read_format_rar_ppmd_use_after_free
 and test_read_format_rar_ppmd_use_after_free2
Origin: upstream, https://github.com/libarchive/libarchive/commit/fa2f626ac0ebe0841370eadd3ffd69679258fa0a
Author: Martin Matuska <martin@matuska.org>
Last-Update: 2019-09-20

--- a/libarchive/test/test_read_format_rar.c
+++ b/libarchive/test/test_read_format_rar.c
@@ -3776,6 +3776,9 @@
   assertA(ARCHIVE_OK == archive_read_next_header(a, &ae));
   assertA(archive_read_data(a, buf, sizeof(buf)) <= 0);
 
+  /* Test EOF */
+  assertA(1 == archive_read_next_header(a, &ae));
+
   assertEqualIntA(a, ARCHIVE_OK, archive_read_close(a));
   assertEqualInt(ARCHIVE_OK, archive_read_free(a));
 }
@@ -3799,6 +3802,9 @@
   assertA(ARCHIVE_OK == archive_read_next_header(a, &ae));
   assertA(archive_read_data(a, buf, sizeof(buf)) <= 0);
 
+  /* Test EOF */
+  assertA(1 == archive_read_next_header(a, &ae));
+
   assertEqualIntA(a, ARCHIVE_OK, archive_read_close(a));
   assertEqualInt(ARCHIVE_OK, archive_read_free(a));
 }
